src/regex.cr

```diff
@@ -240,11 +240,15 @@ class Regex
     # flag that activates both behaviours, so here we do the same by
     # mapping `MULTILINE` to `PCRE_MULTILINE | PCRE_DOTALL`.
     # The same applies for PCRE2 except that the native values are 0x200 and 0x400.
+    # 
+    # For the behaviour of `PCRE_MULTILINE` use `MULTILINE_ONLY`.
 
     # Multiline matching.
     #
     # Equivalent to `MULTILINE | DOTALL` in PCRE and PCRE2.
     MULTILINE = 0x0000_0006
+    # Equivalent to `MULTILINE`in PCRE and PCRE2.
+    MULTILINE_ONLY = 0x0000_0040
```

Comment:
**suggestion:** Theoretically, we're free to chose whatever values we want for this enum because they're specific to Crystal and get translated for any engine implementation. But since all other members are based on the libpcre1 values, we should follow this scheme:

```suggestion
    MULTILINE_ONLY = 0x0000_0002
```

Reply:
> Also this should ensure that MULTILINE_ONLY | DOTALL == MULTILINE which is a useful property.

It's probably worth adding that as an explicit test and/or encoding that directly in the definition of MULTILINE.

Reply:
Oooh the `DOTALL = 2` is surprising. Guess that never worked correctly when it was mapped directly to libpcre1 values 🙈 

I suppose it's probably best to add `MULTILINE_ONLY = 4` then. This breaks value equivalence with libpcre1, but keeps the Crystal values consistent.

Reply:
I tried using the MULTILINE without DOT_ALL value at first but it also breaks inspect (it gets confused and thinks MULTILINE_ONLY is MULTILINE)

Reply:
This should work (somehow). If it doesn't, there's something wrong. Worst case we override the implementation of `CompileOptions#inspect`.

Reply:
Also, changing the values could break existing code, even if I doubt anyone is instantiating an Options[4] or whatever :-D

Reply:
Yeah, it's unlikely. But better to keep this compatibility in case someone relies on the integer values.

Reply:
Please change this to 0x4 then. It'll be the MULTILINE and DOTALL values of PCRE1 swapped.
```suggestion
    MULTILINE_ONLY = 0x0000_0004
```

Reply:
Please change this to 0x4 then. It'll be the MULTILINE and DOTALL values of PCRE1 swapped but still ensure `MULTILINE = MULTILINE_ONLY | DOTALL`.
```suggestion
    MULTILINE_ONLY = 0x0000_0004
```